src/oss-characteristics/CharacteristicTool.cs

```diff
@@ -46,46 +46,48 @@ public class CharacteristicTool
         static void Main(string[] args)
         {
             var characteristicTool = new CharacteristicTool();
-            Logger.Debug($"Microsoft OSS Gadget - {TOOL_NAME} {VERSION}");
+            Logger?.Debug($"Microsoft OSS Gadget - {TOOL_NAME} {VERSION}");
```

Comment:
Do we need to do ?. on Logger? If logger is null, then pretty much all error info is unavailable and we need to know about it

Reply:
Unless you want to crash, yes.

Reply:
Yes, so we check that the Logger is not null during Logger creation, and then assume its not null everywhere else, doing Logger.Warn instead of Logger?.Warn. There is no reason for it to be null anywhere else.
Implicit crashes are not a bad thing in case of catastrophic failures like Out of memory or non working terminal/logging system - since we don't have any way to resolve it, it might be better to crash and let the user know. Wont be practically possible to handle all of them anyways.

Reply:
I'll make logger non-nullable.